torch/nn/modules/module.py

```diff
@@ -241,6 +223,69 @@ def apply(self, fn):
         fn(self)
         return self
 
+    def apply_tensor(self, fn, to_parameters=True, to_gradients=True,
+                     to_buffers=True, recursive=True):
+        r"""Applies ``fn`` to the various tensors within the module.
+
+        This method differs from ``apply()`` in that it acts on the tensors
+        of the module rather than the module itself.
+
+        Specifically, the method will apply the given function to
+        tensor members, after detaching and with version tracking disabled
+        (i.e. on ``.data``).
+
+        Args:
+            fn (:class:`Tensor` -> :class:`Tensor`):
+                function to be applied to each tensor, returning the
+                desired new value of the tensor.
+            to_parameters (bool):
+                whether to apply the function to parameters.
+            to_gradients (bool):
+                whether to apply the function to parameter gradients.
+            to_buffers (bool):
+                whether to apply the function to buffers.
+            recursive (bool):
+                whether to apply the function to tensors in
+                all submodules.
+
+        Returns:
+            Module: self
+
+        Example:
+            # Converts model's non-batchnorm parameters to half precision.
```

Comment:
The example format should be consistent with other examples we have, e.g., prefix with >>> and others. See https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/module.py#L214 for an example.